src/kOS.Safe/Execution/CPU.cs

```diff
@@ -704,16 +695,17 @@ public object GetValue(object testValue, bool barewordOkay = false)
             // If it's a variable, meaning it starts with "$" but
             // does NOT have a value like $<.....>, which are special
             // flags used internally:
-            if (testValue is string &&
-                ((string)testValue).Length > 1 &&
-                ((string)testValue)[0] == '$' &&
-                ((string)testValue)[1] != '<')
+            var identifier = testValue as string;
+            if (identifier == null ||
+                identifier.Length <= 1 ||
+                identifier[0] != '$' ||
+                identifier[1] == '<')
```

Comment:
so far I'm keeping my hands off the cpu class.  I'm planning on merging this PR with mine after I've tested it out, so I'll handle any merges that conflict like this.  The way I have it currently implemented however, this change shouldn't be a problem.  Because I'm using the new methods directly from compile time, only system values will cast as a string (even though `StringValue` has an implicit conversion to string, this is not taken into account by the `as` operator, or so the documentation says).